src/terminal.ml

```diff
@@ -287,24 +287,114 @@ let close_session pid =
 
 let (>>=) = Lwt.bind
 
-let escRemove = Str.regexp
-   ("\\(\\(.\\|[\n\r]\\)+\027\\[[12]J\\)" (* Clear screen *)
-  ^ "\\|\\(\027\\[[0-2]?J\\)" (* Clear screen *)
-  ^ "\\|\\(\027\\[!p\\)" (* Soft reset *)
-  ^ "\\|\\(\027\\][02];[^\007]*\007\\)" (* Set console window title *)
-  ^ "\\|\\(\027\\[\\?25[hl]\\)" (* Show/hide cursor *)
-  ^ "\\|\\(\027\\[[0-9;]*m\\)" (* Formatting *)
-  ^ "\\|\\(\027\\[H\\)") (* Home *)
-
-let escSpace = Str.regexp "\027\\[\\([0-9]*\\)C"
+(* OpenSSH on Windows is known to produce at least the following escape
```

Comment:
While this seems like a big improvement, I wonder if this happens always, or only if TERM supports these.